src/static/js/routes/index.js

```diff
@@ -7,193 +7,244 @@
+  var chartsValueSelector = '.progress-charts_container .chart-value';
+  var chartsDom = document.querySelectorAll( chartsValueSelector );
+  var charts = [];
+  var chartItem;
+  var doneValue;
+
+  for ( var c = 0, len = chartsDom.length; c < len; c++ ) {
+    chartItem = chartsDom[c];
+    doneValue = parseInt( chartItem.innerText || chartItem.textContent, 10 );
+    charts.push( {
+      done:  doneValue / 100,
+      value: 0
     } );
-    // each loop through is a animation frame, so count up the frames
-    iterationCount++;
-    // if charts aren't finished animating, loop again
-    if ( completeCount < 5 ) {
-      window.requestNextAnimationFrame( animate );
-    } else {
-      animationComplete = true;
-    }
   }
 
-  function drawGraph( context, value ) {
-    context.clearRect( 0, 0, context.canvas.width, context.canvas.height );
-    // inner grey
-    context.beginPath();
-    context.arc( center.x, center.y, radius - ( 4 * multiplier ), 0, 2 * Math.PI );
-    context.fillStyle = '#E3E4E5';
-    context.fill();
-
-    var pct;
-    if ( value < 1 ) {
-      // decimal 0.0 - 1 for percentage
-      pct = arcStart + ( value * 360 );
-      // fill percentage arc
-      context.beginPath();
-      context.moveTo( center.x, center.y );
-      context.arc( center.x, center.y, radius, getRadians( arcStart ), getRadians( pct ), false );
-      context.closePath();
-      context.fillStyle = '#FF931B';
-      context.fill();
-      // center circle
-      context.beginPath();
-      context.arc( center.x, center.y, radius / 1.85, 0, 2 * Math.PI );
-      context.fillStyle = '#FFF';
-      context.fill();
-    // switch to green
-    } else {
-      // decimal 0.0 - 1 for percentage
-      pct = arcStart + ( value * 360 );
-      // fill percentage arc
-      context.beginPath();
-      context.moveTo( center.x, center.y );
-      context.arc( center.x, center.y, radius, getRadians( arcStart ), getRadians( pct ), false );
-      context.closePath();
-      context.fillStyle = '#2CB34A';
-      context.fill();
-      // center circle
-      context.beginPath();
-      context.arc( center.x, center.y, radius / 1.85, 0, 2 * Math.PI );
-      context.fillStyle = '#ADDC91';
-      context.fill();
-      // check mark
-      var unit = ( radius - ( 4 * multiplier ) ) / 5;
-      context.beginPath();
-      context.moveTo( center.x - unit, center.y );
-      context.lineTo( center.x - unit / 2, center.y + unit );
-      context.lineWidth = 6 * ( radius / 100 ) * multiplier;
-      context.strokeStyle = '#FFF';
-      context.lineCap = 'round';
-      context.stroke();
-      context.beginPath();
-      context.moveTo( center.x - unit / 2, center.y + unit );
-      context.lineTo( center.x + unit, center.y - unit );
-      context.stroke();
-    }
-  }
+  return charts;
+}
 
+/**
+ * @returns {boolean} Whether requestAnimationFrame is supported or not.
+ */
+function _requestAnimationFrameSupported() { // eslint-disable-line
```

Comment:
I think if a need for second implementation pops up it should, but eventually this whole feature will be removed, so I think keeping it self-contained is okay for now.